### PR TITLE
Add year_of_week and yow Presto functions

### PR DESCRIPTION
**Pull request by kagamiori**

### CODE REVIEW

File: velox/functions/prestosql/DateTimeFunctions.h [hunk 1]
```diff
@@ -208,6 +208,46 @@ struct DayOfYearFunction : public InitSessionTimezone<T> {
   }
 };
 
+template <typename T>
+struct YearOfWeekFunction : public InitSessionTimezone<T> {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  FOLLY_ALWAYS_INLINE int64_t computeYearOfWeek(const std::tm& dateTime) {
+    int isoWeekDay = dateTime.tm_wday == 0 ? 7 : dateTime.tm_wday;
+    // The last a few days in December may belong to the next year if they are
```
Comment (mbasmanova) [inline velox/functions/prestosql/DateTimeFunctions.h:217]: typo: "last a few days" -> "last few days"

File: velox/functions/prestosql/DateTimeFunctions.h [hunk 2]
```diff
@@ -208,6 +208,46 @@ struct DayOfYearFunction : public InitSessionTimezone<T> {
...
+    // in the same week as the next January 1 and this January 1 is a Thursday
+    // or before.
+    if (UNLIKELY(
+            dateTime.tm_mon == 11 && dateTime.tm_mday >= 29 &&
+            dateTime.tm_mday - isoWeekDay >= 31 - 3)) {
+      return 1900 + dateTime.tm_year + 1;
+    }
+    // The first a few days in January may belong to the last year if they are
```
Comment (mbasmanova) [inline velox/functions/prestosql/DateTimeFunctions.h:225]: typo: "first a few days" -> "first few days"

**Review by mbasmanova — APPROVED** [review]
@kagamiori Looks good to me. Thank you for filling in the set of date functions.

File: velox/functions/prestosql/DateTimeFunctions.h (hunk 2 above)
Comment (kagamiori) [reply velox/functions/prestosql/DateTimeFunctions.h]: Thank you for catching these!

**Comment by aditi-pandit** [comment]
> @kagamiori Looks good to me. Thank you for filling in the set of date functions.

+1. Thanks for this.

**Pull request closed without merging**